datafusion/sqllogictest/test_files/order.slt

```diff
@@ -527,9 +527,10 @@ Sort: atan_c11 ASC NULLS LAST
 ----TableScan: aggregate_test_100 projection=[c11]
 physical_plan
 SortPreservingMergeExec: [atan_c11@0 ASC NULLS LAST]
---ProjectionExec: expr=[atan(c11@0) as atan_c11]
-----RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
-------CsvExec: file_groups={1 group: [[WORKSPACE_ROOT/testing/data/csv/aggregate_test_100.csv]]}, projection=[c11], output_ordering=[c11@0 ASC NULLS LAST], has_header=true
+--SortExec: expr=[atan_c11@0 ASC NULLS LAST]
```

Comment:
This test seems to show a regression (there is now a sort in it).
I wonder if the issue is that you need to specify montononicity (which is now possible after https://github.com/apache/arrow-datafusion/pull/9869 from @tinfoil-knight )?